### ISSUE TITLE
[1.6] [Windows] Unable to publish or locally build package

### ISSUE THREAD

**Issue by nathantreid** [meteor/meteor#9287]
# Environment
OS: Windows 10 x64
Meteor: 1.6

# Expected behavior
Running `meteor` with the package installed locally or `meteor publish` from the package's directory should succeed. I am able to publish the package using Linux. The package I'm trying to update/publish is [nathantreid:css-modules](https://github.com/nathantreid/meteor-css-modules).

# Actual behavior
```
C:\projects\atmosphere\meteor-css-modules\meteor-css-modules>meteor publish
mss: updating npm dependencies -- app-module-path, camelcase, cjson, colors, common-tags, css-modules-loader-core, json-to-regex, es6-template-strings, hasha, lru-cache, path-is-absolute, postcss, postcss-modules-local-by-default,
postcss-modules-extract-imports, postcss-modules-scope, postcss-modules-values, ramda, recursive-readdir, shorthash, string-template...
H:\.meteor\packages\meteor-tool\1.6.0\mt-os.windows.x86_32\dev_bundle\lib\node_modules\meteor-promise\promise_server.js:218
      throw error;
      ^

Error: EPERM: operation not permitted, lstat 'C:\projects\atmosphere\meteor-css-modules\meteor-css-modules\.npm\plugin\mss-new-seljuc.zpti\node_modules'
    at Object.fs.lstatSync (fs.js:943:11)
    at Object.lstatSync (H:\.meteor\packages\meteor-tool\1.6.0\mt-os.windows.x86_32\dev_bundle\lib\node_modules\npm\node_modules\graceful-fs\polyfills.js:297:22)
    at rimrafSync (H:\.meteor\packages\meteor-tool\1.6.0\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:290:24)
    at H:\.meteor\packages\meteor-tool\1.6.0\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:341:5
    at Array.forEach (<anonymous>)
    at rmkidsSync (H:\.meteor\packages\meteor-tool\1.6.0\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:340:26)
    at rmdirSync (H:\.meteor\packages\meteor-tool\1.6.0\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:333:7)
    at fixWinEPERMSync (H:\.meteor\packages\meteor-tool\1.6.0\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:208:5)
    at Function.rimrafSync [as sync] (H:\.meteor\packages\meteor-tool\1.6.0\mt-os.windows.x86_32\dev_bundle\lib\node_modules\rimraf\rimraf.js:310:28)
    at Object.files.rm_recursive.path [as rm_recursive] (H:\tools\fs\files.js:307:12)
    at Object.meteorNpm.updateDependencies (H:\tools\isobuild\meteor-npm.js:131:13)
    at Object.compile (H:\tools\isobuild\compiler.js:126:19)
    at Object.buildJsImage (H:\tools\isobuild\bundler.js:3202:26)
    at H:\tools\isobuild\compiler.js:67:33
    at H:\tools\utils\buildmessage.js:359:18
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:352:34
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:350:23
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at Object.enterJob (H:\tools\utils\buildmessage.js:324:26)
    at H:\tools\isobuild\compiler.js:59:18
    at Function._.each._.forEach (H:\.meteor\packages\meteor-tool\1.6.0\mt-os.windows.x86_32\dev_bundle\lib\node_modules\underscore\underscore.js:87:22)
    at Object.compile (H:\tools\isobuild\compiler.js:58:5)
    at H:\tools\isobuild\isopack-cache.js:362:30
    at H:\tools\utils\buildmessage.js:359:18
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:352:34
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:350:23
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at Object.enterJob (H:\tools\utils\buildmessage.js:324:26)
    at IsopackCache._loadLocalPackage (H:\tools\isobuild\isopack-cache.js:317:18)
    at Profile.time (H:\tools\isobuild\isopack-cache.js:251:16)
    at Function.time (H:\tools\tool-env\profile.js:305:10)
    at H:\tools\isobuild\isopack-cache.js:250:17
    at H:\tools\utils\buildmessage.js:359:18
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:352:34
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:350:23
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at Object.enterJob (H:\tools\utils\buildmessage.js:324:26)
    at IsopackCache._ensurePackageLoaded (H:\tools\isobuild\isopack-cache.js:241:20)
    at H:\tools\isobuild\isopack-cache.js:77:14
    at H:\tools\packaging\package-map.js:57:7
    at Function._.each._.forEach (H:\.meteor\packages\meteor-tool\1.6.0\mt-os.windows.x86_32\dev_bundle\lib\node_modules\underscore\underscore.js:87:22)
    at exports.PackageMap.eachPackage (H:\tools\packaging\package-map.js:49:7)
    at IsopackCache.buildLocalPackages (H:\tools\isobuild\isopack-cache.js:76:24)
    at H:\tools\project-context.js:847:25
    at H:\tools\utils\buildmessage.js:359:18
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:352:34
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:350:23
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at Object.enterJob (H:\tools\utils\buildmessage.js:324:26)
    at ProjectContext._buildLocalPackages (H:\tools\project-context.js:846:18)
    at H:\tools\project-context.js:285:9
    at H:\tools\utils\buildmessage.js:359:18
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:352:34
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:350:23
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at Object.enterJob (H:\tools\utils\buildmessage.js:324:26)
    at ProjectContext._completeStagesThrough (H:\tools\project-context.js:275:18)
    at Profile.run (H:\tools\project-context.js:267:12)
    at Function.run (H:\tools\tool-env\profile.js:490:12)
    at ProjectContext.prepareProjectForBuild (H:\tools\project-context.js:266:13)
    at H:\tools\cli\commands-packages.js:402:20
    at H:\tools\utils\buildmessage.js:271:13
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:264:29
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:262:18
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at H:\tools\utils\buildmessage.js:253:23
    at exports.EnvironmentVariable.withValue (H:\tools\utils\fiber-helpers.js:89:14)
    at Object.capture (H:\tools\utils\buildmessage.js:252:19)
    at Object.main.captureAndExit (H:\tools\cli\main.js:274:29)
    at Command.func (H:\tools\cli\commands-packages.js:401:8)
    at H:\tools\cli\main.js:1522:15

```

**Comment by benjamn** [comment]
I agree that publishing from Windows should work, and this error is definitely some kind of bug. However, since you were able to publish on Linux, are you then able to `meteor add nathantreid:css-modules` on Windows (without publishing from Windows)? As long as the Windows machine can recompile binary npm packages, you shouldn't have to publish from Windows at all. At least that's my understanding/intention.

**Comment by nathantreid** [comment]
Yeah, that works - I just have my main dev environment set up on Windows right now, so it's easier to work on updates from there.

**Comment by benjamn** [comment]
Kinda sad that the stack trace shows a function called `fixWinEPERMSync` and yet the error is `EPERM` anyway.

**Comment by ExcelNet-Public** [comment]
Is any work being done on this issue, believe I might be facing it as well as we have local packages being used?

**Comment by hwillson** [comment]
Work on addressing this has been scheduled for Meteor 1.6.2 (see the related issue https://github.com/meteor/meteor/issues/9540).

**Comment by hwillson** [comment]
https://github.com/meteor/meteor/commit/9e60539f113af2f6dea5b2fed5323bd27114c4ac was included in release 1.6.1, and should help address this issue. Please let us know if it's still happening. Thanks!
